Adding transfer operation

rranjan3 opened the pull request:
Updates in CLI & java TP to handle transfer of amount from one wallet to another

askmish commented on javaprocessor/src/main/java/simplewallet/processor/SimpleWalletProcessor.java:
```diff
@@ -212,6 +214,64 @@ private void makeWithdraw(State stateInfo, String operation, Integer amount, Str
+	if (balance.isEmpty()) {
+	    throw new InvalidTransactionException("Didn't find the wallet key associated with user key " + userFromKey);
+	}
+	Integer value = Integer.valueOf(balance);
```
Please use proper variable names - balance, values, amount :)

askmish commented on javaprocessor/src/main/java/simplewallet/processor/SimpleWalletProcessor.java:
```diff
@@ -212,6 +214,64 @@ private void makeWithdraw(State stateInfo, String operation, Integer amount, Str
+	logger.info("Debitting balance with " + amount);
+	stateInfo.setState(newLedgerEntry);
+	
+	//Crediting to creditor
```
All validations should be done prior to money transfer.

askmish commented on pyclient/wallet/simplewallet_client.py:
```diff
@@ -82,9 +82,30 @@ def deposit(self, value):
             "withdraw",
             value)
+            #raise SimpleWalletException('Encountered an error during withdrawal {}'.format(retValue) )
+        except Exception as e:
```
Why this exception.

askmish commented on pyclient/wallet/simplewallet_client.py:
```diff
@@ -130,21 +151,29 @@ def _send_to_restapi(self,
         # Construct the address where we'll store our state
         address = self._address
         
+        toAddress = address if "transfer" != action else (_hash(FAMILY_NAME.encode('utf-8'))[0:6] + \
```
instead of the existing logic, create a separate address list variable and add/remove addresses as required.

rranjan3 committed:
Adding transfer operation with support in Java TP

Updates in CLI & java TP to handle transfer of amount from one wallet to another
